### ISSUE TITLE
nvim-web-devicons error.

### ISSUE THREAD

**Issue by DMunkei** [sindrets/diffview.nvim#298]
I saw in the changelog `changelog-271` that changes to the settings had to be done. I didn't have anything special in my settings, just the standard example written in the `README`. I simply exchanged it with the current example in the `README`. However, I am getting this error:
`[Diffview.nvim] nvim-web-devicons is required to use file icons! Set "use_icons = false" in your config to stop seeing this message.`
I have the dependency installed, not sure what's the issue here?

**Comment by sindrets** [comment]
Please show the output from `:checkhealth diffview`.

**Comment by DMunkei** [comment]
```
diffview: require("diffview.health").check()
========================================================================
## Checking plugin dependencies
  - OK: plenary installed.
  - WARNING: Optional dependency 'nvim-web-devicons' not found.

## Checking external dependencies
  - OK: Git found.
  - OK: Git is up-to-date.
```
Here's also the line in my `packer.lua` that is pulling the dependency
```lua
  use {'nvim-lualine/lualine.nvim',
      requires = { 'kyazdani42/nvim-web-devicons', opt = true }
    }
```

**Comment by sindrets** [comment]
You're pulling it as an `opt` dependency. Thus it will not be available on the `'runtimepath'` until either you, or packer add it with `:packadd`.

Either don't install it as `opt`, or declare it as a dependency in diffview's packer spec as well.

**Comment by DMunkei** [comment]
I tried doing this :
```lua
  use { 'sindrets/diffview.nvim',
  requires = {'nvim-lua/plenary.nvim', 'kyazdani42/nvim-web-devicons', opt = true}
  }
``` 
However, that didn't seem to work. I tried doing it like you mentioned with `:packadd nvim-web-devicons` and then opened `DiffViewOpen`. Then that worked. My question is, where would I add `:packadd nvim-web-devicons`?

**Comment by sindrets** [comment]
It's a library. I don't understand why you want to lazy load it. It doesn't do anything until another plugin calls its API.

If, for whatever reason, it's really important to you to lazy load it, then I'll leave it up to you to figure out when it should be `:packadd`-ed. It has to happen at least before the first time you call any of diffview's commands.

It's a lot simpler if you don't care about lazy loading it:

```lua
use {
  "sindrets/diffview.nvim",
  requires = {
    "nvim-lua/plenary.nvim",
    "kyazdani42/nvim-web-devicons",
  },
}
```

**Comment by DMunkei** [comment]
No reason at all, I took it from the package docs. I just changed it to your proposed suggestion. However, it still requires me to do a manual `:packadd nvim-web-devicons` for it to work.  I had to also remove the `opt = true` in the other import I had. Works properly now.
```lua
 use { 'nvim-lualine/lualine.nvim',
        requires = { 'kyazdani42/nvim-web-devicons'}
    }
    use {
        'sindrets/diffview.nvim',
        requires = {
            'nvim-lua/plenary.nvim',
            'kyazdani42/nvim-web-devicons'
        },
    }
```

**Comment by sindrets** [comment]
Right. A packer dependency can't be declared multiple times with different specs. That seems like a limitation in packer. You could raise that issue upstream if you want it addressed.

Closing, as the problem has been solved.

**Comment by tummetott** [comment]
@DMunkei btw the repo is called `nvim-tree/nvim-web-devicons` now. The old name still links to the repo but just to be explicit, I'd use the new repo name

**Comment by DMunkei** [comment]
@tummetott Thanks, changed it.
